index.js

```diff
@@ -0,0 +1,89 @@
+const baseURL = "https://ghibliapi.herokuapp.com/films"
+
+document.addEventListener("DOMContentLoaded", async () => {
+    let allMoviesData = await getAllMovies();
+    let selectedMovie;
+
+    addMoviesTitlesToSelections (allMoviesData);
+
+    let SelecMenu = document.querySelector("#selectBox");
+    SelecMenu.addEventListener("change", () => {
+
+        let MoviesList = document.querySelectorAll("option");
+
+        if (MoviesList[0].selected) {
+            selectedMovie = null;
+            return; // USING THIS TO PREVENT REVIEW WHEN NO SELECTION
+        }
+
+        for (let i = 1; i < MoviesList.length; i++) {
+
+            if (MoviesList[i].selected) {
+                selectedMovie = MoviesList[i].innerText;
+                getInfoAboutMovie(allMoviesData, selectedMovie);
+                break;
+            }
+        }
+    }) // END OF EVENT LISTNER FOR SELECTION
+
+    let form = document.querySelector("form");
+    form.addEventListener("submit", (event) => {
+        event.preventDefault();
+
+        let title = document.querySelector("#title").innerText;
+        let reviewInput = form.querySelector("input[type='text']");
+        
+        if (selectedMovie && reviewInput.value) {
+            let reviewsContainer = document.querySelector("#reviews");
+
+            let newReview = document.createElement("li");
+            newReview.innerHTML = `<strong>${title}:</strong> ${reviewInput.value}`;
+            reviewsContainer.appendChild(newReview);
+            reviewInput.value = "";
+        }
+    })
+
+
+}) // DOM CONTENT LOADED
+
+const getAllMovies = async () => {
+    let onlineResponse = await axios.get(baseURL)
+                                    .catch (error => {
+                                        console.log(error)
+                                    })
+    return onlineResponse.data;
+}
+
+const addMoviesTitlesToSelections = (fullMoviesInfo) => {
+    let SelecMenu = document.querySelector("#selectBox");
+    
+    for (let movie of fullMoviesInfo) {
+        let newOption = document.createElement("option");
+        newOption.value = movie.title;
+        newOption.innerText = movie.title;
+        SelecMenu.appendChild(newOption);
+    }
+}
+
+const getInfoAboutMovie = (allFilms, title) => {
+    for (let movie of allFilms) {
+        if (movie.title === title) {
+            addMovieToDOM(movie);
+            break;
+        }
+    }
```

Comment:
Besides the better approach that I mentioned above, our intent here was for you to make a network request to `/films/<film-id>` after grabbing the id of the selected movie to get the details of the movie. Good thing you wrote a separate function for this, how would you have change this to make network request?

Reply:
I just thought it could be faster
to loop over 20 elements then making an other network request (where
internet speed could be variable)
I first thought os using what you suggested then opted for the
second option (loop)

I fixed my code as well making an other network request

Reply:
This is definitely a valid point. However part the assessment was for us to see if you all are able make multiple network requests.

Reply:
completely understood that, I just didn't see it that way since it was not asked straight forward on the assessment

Reply:
Making multiple network requests is not a problem for me (I believe), you can see that in my previous lab (the pokemon game) where I have 12 network requests in total (using 2 fetch requests in two different functions if I remember well)